openhcl/openhcl_boot/src/host_params/dt/mod.rs

```diff
@@ -375,6 +375,17 @@ struct PartitionTopology {
     memory_allocation_mode: MemoryAllocationMode,
 }
 
+// Calculate the default mmio size for VTL2 when not specified by the host.
+//
+// This is half of the high mmio gap size, rounded down, with a minimum of 128
+// MB and a maximum of 1 GB.
+fn calculate_default_mmio_size(parsed: &ParsedDt) -> Result<u64, DtError> {
+    const MINIMUM_MMIO_SIZE: u64 = 128 * (1 << 20);
+    const MAXIMUM_MMIO_SIZE: u64 = 1 << 30;
+    let half_high_gap = parsed.vmbus_vtl0.as_ref().ok_or(DtError::Vtl0Vmbus)?.mmio[1].len() / 2;
+    Ok(half_high_gap.clamp(MINIMUM_MMIO_SIZE, MAXIMUM_MMIO_SIZE))
+}
+
```

Comment:
got a test for this?

Reply:
well, i can, but it's not that straightforward.

Reply:
Note that we _really_ want the host to specify the value to allocate here. I'll see about getting that change plumbed thru on the hyper-v side (it's not, today).